flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -383,25 +384,28 @@ private void connect(Integer headOfChain, StreamEdge edge) {
 		downStreamConfig.setNumberOfInputs(downStreamConfig.getNumberOfInputs() + 1);
 
 		StreamPartitioner<?> partitioner = edge.getPartitioner();
+		JobEdge jobEdge = null;
 		if (partitioner instanceof ForwardPartitioner) {
-			downStreamVertex.connectNewDataSetAsInput(
+			jobEdge = downStreamVertex.connectNewDataSetAsInput(
 				headVertex,
 				DistributionPattern.POINTWISE,
 				ResultPartitionType.PIPELINED,
 				true);
 		} else if (partitioner instanceof RescalePartitioner){
-			downStreamVertex.connectNewDataSetAsInput(
+			jobEdge = downStreamVertex.connectNewDataSetAsInput(
 				headVertex,
 				DistributionPattern.POINTWISE,
 				ResultPartitionType.PIPELINED,
 				true);
 		} else {
-			downStreamVertex.connectNewDataSetAsInput(
+			jobEdge = downStreamVertex.connectNewDataSetAsInput(
 					headVertex,
 					DistributionPattern.ALL_TO_ALL,
 					ResultPartitionType.PIPELINED,
 					true);
 		}
+		// set strategy name so that web interface can show it.
+		jobEdge.setShipStrategyName(partitioner.toString());
```

Comment:
Yes.  The plan is generated by the `JsonPlanGenerator`, which picks up the ship strategy.